x-pack/packages/ml/query_utils/src/filter_based_simple_query.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isPopulatedObject } from '@kbn/ml-is-populated-object';
+
+import { isSimpleQuery } from './simple_query';
+import type { FilterBasedSimpleQuery } from './types';
+
+export function isFilterBasedSimpleQuery(arg: unknown): arg is FilterBasedSimpleQuery {
+  return (
+    isPopulatedObject(arg, ['bool']) &&
+    isPopulatedObject(arg.bool, ['filter']) &&
```

Comment:
just to clarify, is it intentional that this function does not check for other keys in the `bool` query?

Reply:
In this case yes, we're just interested of a simple query within the filter, not the others options like `must` and others. I tried to clarify this a bit more in the code, did a bit of renaming and added more comments. The comments include that the purpose of these helpers is to identify stuff in the bool queries the Kibana/EUI search bars expose.